packages/terminal/src/browser/terminal-keybinding-contexts.ts

```diff
@@ -33,3 +34,19 @@ export class TerminalActiveContext implements KeybindingContext {
         return this.shell.activeWidget instanceof TerminalWidget;
     }
 }
+
+@injectable()
+export class TerminalHideSearchContext implements KeybindingContext {
+    readonly id: string = TerminalKeybindingContexts.terminalHideSearch;
+
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    isEnabled(): boolean {
+        if (!(this.shell.activeWidget instanceof TerminalWidget)) {
+            return false;
+        }
+        const searchWidget = (this.shell.activeWidget as TerminalWidget).getSearchBox();
```

Comment:
(nit) typescript can infer type based on type guards, it knows that it is `TerminalWidget` already